prow/cmd/branchprotector/protect.go

```diff
@@ -188,51 +191,70 @@ func (p *Protector) ConfigureBranches() {
 func (p *Protector) Protect() {
 	bp := p.cfg.BranchProtection
 
-	// Scan the branch-protection configuration
-	for orgName, org := range bp.Orgs {
-		if err := p.UpdateOrg(orgName, org, bp.Protect, bp.Contexts, bp.Pushers); err != nil {
-			p.errors.add(err)
-		}
-	}
+	// TODO(fejta): check for deprecated usage and warn
+	policy := bp.MakePolicy()
 
-	// Do not automatically protect tested repositories
-	if !bp.ProtectTested {
-		return
-	}
+	if bp.ProtectTested { // Automatically require any required prow jobs for unconfigured repos
+		for repoPath := range p.cfg.Presubmits {
+			parts := strings.Split(repoPath, "/")
+			if len(parts) != 2 {
+				p.errors.add(fmt.Errorf("bad repo in prow jobs: %s", repoPath))
+				continue
+			}
+			orgName := parts[0]
+			repoName := parts[1]
+			repoReqs := repoRequirements(orgName, repoName, *p.cfg)
```

Comment:
Can't we partition this by branch by looking at the prow job configs?

Reply:
Done